rotkehlchen/db/drivers/gevent.py

```diff
@@ -366,12 +366,19 @@ def write_ctx(self, commit_ts: bool = False) -> Generator['DBCursor', None, None
                 raise
             else:
                 if commit_ts is True:
-                    cursor.execute(
-                        'INSERT OR REPLACE INTO settings(name, value) VALUES(?, ?)',
-                        ('last_write_ts', str(ts_now())),
-                    )
-                    # last_write_ts in not cached to cached settings. This is a critical section
-                    # and adding even one more function call can have very ugly and
+                    try:
+                        cursor.execute(
+                            'INSERT OR REPLACE INTO key_value_cache(name, value) VALUES(?, ?)',
+                            ('last_write_ts', str(ts_now())),
+                        )
+                    except sqlcipher.OperationalError:  # pylint: disable=no-member
+                        cursor.execute(  # when key_value_cache is missing go back to settings
+                            # it happens while upgrading
```

Comment:
leave the comment in one line or add it over the cursor.execute statement. Also Maybe add that this is needed as long as we have v40->v41 upgrade